src/core/symbology/qgslinesymbollayer.cpp

```diff
@@ -2420,6 +2551,195 @@ void QgsTemplatedLineSymbolLayerBase::renderPolylineCentral( const QPolygonF &po
   }
 }
 
+QList<QList<QgsTemplatedLineSymbolLayerBase::BlankSegments>> QgsTemplatedLineSymbolLayerBase::parseBlankSegments( const QString &strBlankSegments, const QgsRenderContext &renderContext, Qgis::RenderUnit unit, QString &error )
```

Comment:
Here is the real BlankSegmentParser :) I think you should externalise it as a class: it can make the code clearer and the lambda functions will be private members

Reply:
I hesitated on this one too, and that's maybe better. I'll do it